File: 300 Longest Increasinf Subseq/LIS.cpp
#include <iostream>
#include <vector>
using namespace std;


int LIS(vector<int> nums){
    int n = nums.size();
    vector<int> lis(n,1) ;
    for (int i = 1;i<n;i++){
        for (int j =0;j<i;j++){
            if(nums[j]<nums[i]){
                lis[i] = max(lis[i],lis[j]+1);
            }
        }
    }
    return *max_element(lis.begin(),lis.end());
}

int main(){
    vector<int> nums;
    nums.push_back(3); 
    nums.push_back(4); 
    nums.push_back(2); 
    nums.push_back(8); 
    nums.push_back(10); 
    nums.push_back(5); 
    nums.push_back(1); 
    cout<<LIS(nums);
}

File: 300 Longest Increasinf Subseq/LIS.rs
use std::cmp;


fn LIS(nums: Vec<i32>,n:i32) -> i32{
    let mut lis = vec![1];
    for i in (1..n){
        lis[i] = 1;
        for j in (0..*i){
            if (nums[j] < nums[i]){
                lis[i] = cmp::max(lis[i],lis[j]+1);
            }
        }
    return lis.iter().max();
    }
}

fn main(){

    let num = vec![3,4,2,8,10,5,1]; 
    println!("{}",LIS(num,7));
}
